src/page/home.tsx

```diff
@@ -0,0 +1,76 @@
+import { useState, useEffect } from "react";
+import Todos from "../components/todos";
+import InputForm from "../components/input.form";
+import { nanoid } from "nanoid";
+import { todoType } from "../components/types/todo.type";
+const Home = () => {
+  const initialTodos: todoType[] = JSON.parse(
+    localStorage.getItem("todos") || "[]"
+  );
+  const [todos, setTodos] = useState(initialTodos);
+  const [formData, setFormData] = useState({ todo: "" });
+  const [isButtonDisabled, setIsButtonDisabled] = useState(true);
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setFormData((previousFormData) => {
+      return {
+        ...previousFormData,
+        [event.target.name]: event.target.value,
+      };
+    });
+    setIsButtonDisabled(event.target.value.trim() === "");
+  };
+
+  const createTodos = (): todoType => {
+    return {
+      id: nanoid(),
+      todo: formData.todo,
+      isActive: true,
+    };
+  };
+
+  const submitForm = () => {
+    setTodos((previous: todoType[]) => [createTodos(), ...previous]);
+    setFormData({ todo: "" });
+    setIsButtonDisabled(true);
+  };
```

Comment:
Do these functions need to be in this file? Since they are only related to the form component and are not being shared anywhere else why not put them in the form component.

Reply:
They are also affecting even the todos component while we are submitting, So I thought maybe this file can hold functions and all imported components other component will hold only the returned component to render with props

Reply:
Okay, that makes sense. Another alternative would be to pass the `setState` function in the form component and move everything else into the FormComponent